### PR TITLE
Add xor field validation for models

### PR DESCRIPTION
**Pull request by vtemian**
Since the hierarchy of models is too complex, I suggest moving the validation from form to model layers.
The form implementation is not error-prone or easy to maintain since the ModelForm will need to know fields from certain parent models.

For example [TranscribeOwnedIncomeFromOtherSourcesRowEntry](https://github.com/code4romania/catpol-declaratii/blob/develop/project_template/forms.py#L335), needs to know about `commune` and `city`, that are not defined neither in [OwnedIncomeFromOtherSourcesTableEntry](https://github.com/code4romania/catpol-declaratii/blob/develop/project_template/models.py#L404), not in [CommonIncomeFields](https://github.com/code4romania/catpol-declaratii/blob/develop/project_template/models.py#L115), but in [CommonInfo](https://github.com/code4romania/catpol-declaratii/blob/develop/project_template/models.py#L105).

Given that, moving validation to the model layer is easier to understand and maintain. The only problem is that model's `full_clean` method [is not called on save](https://docs.djangoproject.com/en/3.0/ref/models/instances/#validating-objects). In order to fix it, I've added the `AutoCleanMixin`. 

Going further, to declare a bunch of fields that are not meant to be filled together, one just needs to define an attribute containing a list of dicts (each dict representing a group of fields from that model that needs to be xor-ed, in case a model has more than two), with the keys representing the field's name and the values representing the corresponding translation.
```python
XOR_FIELDS = [
    {
        "commune": _("commune"),
        "city": _("city"),
    }
]
```
That attribute is checked in `XORModelMixin`'s `full_clean` method and if all of the fields are `None` or `''` or more than one field is not, then it throws a `ValidationError` that got propagated and rendered in the task's template.

### CODE REVIEW

File: project_template/models.py [hunk 1]
```diff
@@ -51,7 +52,7 @@ def validate_percentage(value):
         )
 
 
-class Politician(models.Model):
+class Politician(AutoCleanModelMixin, models.Model):
```
Comment (catileptic) [inline project_template/models.py]: Why did you add the Mixin to this model class?

File: project_template/models.py [hunk 2]
```diff
@@ -81,7 +82,7 @@ def __full_name(self):
         return "{} {}".format(self.name, self.surname)
 
 
-class Declaration(DocumentModel):
+class Declaration(AutoCleanModelMixin, DocumentModel):
```
Comment (catileptic) [inline project_template/models.py]: Why did you add the Mixin to this model class?

File: project_template/models.py [hunk 3]
```diff
@@ -95,14 +96,21 @@ def __str__(self):
         )
 
 
-class Person(models.Model):
+class Person(AutoCleanModelMixin, models.Model):
```
Comment (catileptic) [inline project_template/models.py]: Why did you add the Mixin to this model class?

File: project_template/models.py [hunk 4]
```diff
@@ -402,5 +417,5 @@ class OwnedIncomeFromOtherSourcesTable(models.Model):
 
 # Tabel Venituri din alte surse - actual row information
 class OwnedIncomeFromOtherSourcesTableEntry(CommonIncomeFields):
-    table = models.ForeignKey(OwnedIncomeFromOtherSourcesTable, on_delete=models.CASCADE, null=True)
+    table = models.ForeignKey(OwnedIncomeFromOtherSourcesTable, on_delete=models.CASCADE, null=True, blank=True)
```
Comment (catileptic) [inline project_template/models.py]: Please remove `blank=True`. You probably used it for testing purposes.

File: project_template/utils.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            return
+
+        super().full_clean(*args, **kwargs)
+
+        self.is_cleaned = True
+
+
+class XORModelMixin:
+    XOR_FIELDS = [
+        # {
```
Comment (catileptic) [inline project_template/utils.py]: Please remove the comments.

Comment (vtemian) [reply project_template/utils.py]: I thought that there are useful for documentation. I'll add a proper docstring for it, instead.

Comment (catileptic) [reply project_template/utils.py]: Hm, that's a fair point. Yeah, I think the comment does a better job at exemplifying the possible values.

**Comment by catileptic** [comment]
Thank you for removing `tests_tasks.py` once and for all. 

Tests are in order (and much appreciated). 

Would you be up to adding a GH action that runs them?

**Comment by vtemian** [comment]
@catileptic done ^_^

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+name: CI
+
+on: [push, pull_request]
```
Comment (RaduCStefanescu) [inline .github/workflows/tests.yml]: I would keep the tests running only on `pull_request` on `develop` branch

Comment (vtemian) [reply .github/workflows/tests.yml]: Done @RaduCStefanescu
